polygeist/tools/cgeist/Test/Verification/sycl/function_filter.cpp

```diff
@@ -0,0 +1,37 @@
+// RUN: clang++ -fsycl -fsycl-device-only -emit-mlir -Xcgeist -function=match %s -o - | FileCheck %s
+
+// COM: No GPU functions.
+// CHECK-NOT: gpu.func
+
+// COM: No definitions with name contain func.
+// CHCEK-NOT: func.func {{.*}}func{{.*}} {
+
+// COM: func1 is a declaration.
+// CHECK: func.func private @func1()
+
+// COM: match1 and match2 are kept as definitions.
+// CHECK: func.func @match1() {{.*}} {
```

Comment:
So the `-function=match` will cause any `.*match.*` to be kept rather than only the exact match ? 
I'm not sure about that. Would it be possible to match the exact name or for the user to give a regexp to use for matching ?

Reply:
Answered on the other comment.